### DISCUSSION TITLE
rerunning flux bootstrap github --read-write-key doesn't recreate a read/write deploy key

### DISCUSSION THREAD

**Discussion by cer** [fluxcd/flux2#3862] [Q&A] [answered]
Hi,

I wanted add image update automation to an existing cluster so I added the needed options and reran:

```
flux bootstrap github \
  --components-extra=image-reflector-controller,image-automation-controller \
  --read-write-key \
```

The components were updated but the deploy key was still read-only.
As a result, I got this error:

`  Warning  error   67s (x13 over 29m)  image-automation-controller  unknown error: ERROR: The key you are authenticating with has been marked as read only.`

Is this the intended behavior?

What's the solution? `flux uninstall` followed by `flux bootstrap`?

**Comment by stefanprodan** [comment]
> Is this the intended behavior?

Yes, we don't touch existing Git secrets.

> What's the solution?

Delete the `flux-system` secret before running bootstrap.

**Reply by makkes** [reply, marked as answer]
I created https://github.com/fluxcd/website/pull/1496 to clarify the semantics around the deploy key. Hope it helps.

Edit: Now documented at https://fluxcd.io/flux/installation/#deploy-key-rotation.
